src/components/projects-summary/projects-summary.controller.ts

```diff
@@ -135,27 +135,9 @@ export class ProjectsSummaryController implements angular.IController {
   };
 
   public $onChanges(onChangesObj: angular.IOnChangesObject) {
-    if (onChangesObj.serviceClasses || onChangesObj.imageStreams) {
-      if (!this.ctrl.serviceClasses && !this.ctrl.imageStreams) {
-        return;
-      }
-
-      // Convert service classes to ServiceItem, which is needed for the ordering panel.
-      let items = _.map(this.ctrl.serviceClasses, (serviceClass) => {
-        return this.Catalog.getServiceItem(serviceClass);
-      });
-
-      // Convert builders to ImageItem.
-      items = items.concat(_.map(this.ctrl.imageStreams, (imageStream) => {
-        return this.Catalog.getImageItem(imageStream);
-      }));
-
-      // Remove null items (non-builder images).
-      items = _.reject(items, (item) => {
-        return !item;
-      });
-
-      this.allItems = _.indexBy(items, 'resource.metadata.uid');
+    if (onChangesObj.catalogServiceItems && (!onChangesObj.catalogServiceItems.isFirstChange() ||
+      !_.isEmpty(onChangesObj.catalogServiceItems.currentValue))) {
```

Comment:
Same comment here. `null` or `undefined` if not loaded, and we should be able to remove the `isFirstChange` logic. I don't think we should assume what first change and subsequent changes mean